Sources/WalletConnectSign/Types/Session/WCSession.swift

```diff
@@ -107,30 +113,33 @@ struct WCSession: ExpirableSequence {
         return false
     }
 
-    mutating func updateNamespaces(_ namespaces: [String: SessionNamespace]) {
+    mutating func updateNamespaces(_ namespaces: [String: SessionNamespace], timestamp: Date = Date()) {
         self.namespaces = namespaces
+        self.timestamp = timestamp
     }
 
     /// updates session expiry by given ttl
     /// - Parameter ttl: time the session expiry should be updated by - in seconds
-    mutating func updateExpiry(by ttl: Int64) throws {
+    mutating func updateExpiry(by ttl: Int64, timestamp: Date) throws {
         let newExpiryDate = Date(timeIntervalSinceNow: TimeInterval(ttl))
         let maxExpiryDate = Date(timeIntervalSinceNow: TimeInterval(WCSession.defaultTimeToLive))
         guard newExpiryDate > expiryDate && newExpiryDate <= maxExpiryDate else {
             throw WalletConnectError.invalidUpdateExpiryValue
         }
-        expiryDate = newExpiryDate
+        self.expiryDate = newExpiryDate
+        self.timestamp = timestamp
     }
 
     /// updates session expiry to given timestamp
     /// - Parameter expiry: timestamp in the future in seconds
-    mutating func updateExpiry(to expiry: Int64) throws {
+    mutating func updateExpiry(to expiry: Int64, timestamp: Date) throws {
         let newExpiryDate = Date(timeIntervalSince1970: TimeInterval(expiry))
         let maxExpiryDate = Date(timeIntervalSinceNow: TimeInterval(WCSession.defaultTimeToLive))
         guard newExpiryDate > expiryDate && newExpiryDate <= maxExpiryDate else {
             throw WalletConnectError.invalidUpdateExpiryValue
         }
-        expiryDate = newExpiryDate
+        self.expiryDate = newExpiryDate
+        self.timestamp = timestamp
```

Comment:
so you decided to have one timestamp for `extend` and `update`....

Reply:
Removed this logic for extendSession